### ISSUE TITLE
Missing List Object

### ISSUE THREAD

**Issue by midnightconman** [jsonnet-libs/k8s#9]
ksonnet includes a generic list object in `k.libsonnet` that can be used to combine multiple objects into a single object to send to a kubernetes api server. The implementation for this list object is [here](https://github.com/ksonnet/ksonnet-lib/blob/master/ksonnet.beta.4/k.libsonnet#L85).

The `List` type can be found [here](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L914) for kubernetes.

**Comment by sh0rez** [comment]
Hi! Thanks for raising this issue.

While designing the generator of this project, we omitted `List` types on purpose, as there were quite a lot of them which were all the same but had different names. 
We also experienced `List` to be sort-of an antipattern, because it makes analyzing the resources inside the list harder, without adding any benefits afaict.

From a API and UX perspective, it might still make sense to have a single generic List type, which could be defined in `custom/list.libsonnet`. 

Could you give an example of a use-case where `List` is required / much better than alternatives?

**Comment by midnightconman** [comment]
My setup is a little bit different, as I don't use the ksonnet or tanka binaries. I use bazel for all my apps, so naturally I use bazel to bridge the gap between jsonnet and kubernetes. I use a bazel library called [rules_k8s](https://github.com/bazelbuild/rules_k8s) to interface with kubernetes and bazel's [rules_jsonnet](https://github.com/bazelbuild/rules_jsonnet) to transform k.libsonnet into json kubernetes can understand. In `rules_k8s`, there is a function called `k8s_object`, the normal way this would work is that you would say add a `deployment` object as a json / yaml manifest and then you can use bazel commands to apply / delete the object in kubernetes. `k8s_object` looks for a valid object and it turns out there is a valid k8s meta object -- `List`.

So long story short, in a jsonnet file I create a local variable using the normal import and merge mechanisms in jsonnet. Then I use a list object to select which manifest objects I would like to use in my `k8s_object`. Here is an example of the jsonnet:

```
local k = import 'k.libsonnet';

local m = (import 'memcached.libsonnet') + (import 'configs.libsonnet') +
  {
    _config+: { var: true },
  };

k.core.v1.list.new([
  m.service,
  m.deployment,
]),
```

It makes sense... I don't think people would use a list of roleBindings in a jsonnet / libsonnet to combine multiple roleBindings :shrug:. But I do think there is merit to having the core List object.

I only use the core List type (so far). I use `list.new()` and `list.items()`.

**Comment by sh0rez** [comment]
Right, that sounds very reasonable to me.

Do you want to contribute this feature yourself? You'd need to create a file called `custom/list.libsonnet` and add the custom type in there as such:

```jsonnet
{
  core+: {
    v1+: {
      list+: {
        new(items):: { /* ... */} 
      }
    }
  }
}

**Comment by midnightconman** [comment]
That works for me... I'll put in a PR. :smile_cat: Thank you
